src/Kafka/ManageKafkaPermissions/components/ManagePermissionsModal.tsx

```diff
@@ -0,0 +1,94 @@
+import {
+  Button,
+  Form,
+  FormGroup,
+  Modal,
+  SelectOptionObject,
+} from "@patternfly/react-core";
+import { useRef } from "react";
+import { useTranslation } from "react-i18next";
+import { SubmitButton } from "./PermissionsModalSubmitButton";
+import { Validated, SelectAccount } from "./SelectAccount";
+
+export enum PrincipalType {
+  UserAccount = "USER_ACCOUNT",
+  ServiceAccount = "SERVICE_ACCOUNT",
+}
+
+export type Principal = {
+  id: string;
+  principalType?: PrincipalType;
+  displayName?: string;
+};
```

Comment:
These types should live in a `types.ts` file, probably a level above the components folder.

I'm not sure about the use of the word `Principal` here. In the UI we show this as an `Account`, we should stick to that name.

```suggestion
export enum AccountType {
  User = "USER_ACCOUNT",
  Service = "SERVICE_ACCOUNT",
}

export type Account = {
  id: string;
  type: AccountType;
  displayName?: string;
};
```

I also commented in the past in some other PR that [enums aren't a great choice](https://blog.logrocket.com/why-typescript-enums-suck/) and I stand my point, although it's not a hill I'd die on.